src/V3EmitCSyms.cpp

```diff
@@ -235,19 +238,28 @@ class EmitCSyms final : EmitCBaseVisitor {
                     }
                     // UINFO(9, "For " << scopep->name() << " - " << varp->name() << "  Scp "
                     // << scpName << "Var " << varBase << endl);
-                    const string varBasePretty = AstNode::prettyName(VName::dehash(varBase));
-                    const string scpPretty = AstNode::prettyName(VName::dehash(scpName));
-                    const string scpSym = scopeSymString(VName::dehash(scpName));
-                    // UINFO(9, " scnameins sp " << scpName << " sp " << scpPretty << " ss "
-                    // << scpSym << endl);
-                    if (v3Global.opt.vpi()) varHierarchyScopes(scpName);
-                    if (m_scopeNames.find(scpSym) == m_scopeNames.end()) {
-                        m_scopeNames.insert(std::make_pair(
-                            scpSym, ScopeData{scpSym, scpPretty, 0, "SCOPE_OTHER"}));
+
+                    if (varp->isSigUserRdPublic()) {
```

Comment:
So what was happening was it was only calling varHierarchyScopes on modules that had public signals in them, and so only copying over those orig names and SCOPE_MODULE types. Now this will call it on those, but not add any new scopes if they don't exist. I'm worried this may impact build times though by iterating through more signals, so I'd recommend benchmarking before merging. @wsnyder

Reply:
Actually noticing quite a bit of build time impact, I'll change this so it just adds the modules and not a bunch of signals